internal/impl/postgresql/integration_test.go

```diff
@@ -0,0 +1,942 @@
+		return len(outBatches) == 1000
+	}, time.Second*25, time.Millisecond*100)
+
+	for i := 0; i < 1000; i++ {
+		_, err = db.Exec("INSERT INTO flights (name, created_at) VALUES ($1, $2);", fake.Address().City(), fake.Time().RFC1123(time.Now()))
+		_, err = db.Exec("INSERT INTO flights_non_streamed (name, created_at) VALUES ($1, $2);", fake.Address().City(), fake.Time().RFC1123(time.Now()))
+		require.NoError(t, err)
+	}
+
+	assert.Eventually(t, func() bool {
+		outBatchMut.Lock()
+		defer outBatchMut.Unlock()
+		return len(outBatches) == 2000
+	}, time.Second, time.Millisecond*100)
+
+	require.NoError(t, streamOut.StopWithin(time.Second*10))
+
+	// Starting stream for the same replication slot should continue from the last LSN
+	// Meaning we must not receive any old messages again
+
+	streamOutBuilder = service.NewStreamBuilder()
+	require.NoError(t, streamOutBuilder.SetLoggerYAML(`level: OFF`))
+	require.NoError(t, streamOutBuilder.AddCacheYAML(cacheConf))
+	require.NoError(t, streamOutBuilder.AddInputYAML(template))
+
+	outBatches = []string{}
+	require.NoError(t, streamOutBuilder.AddBatchConsumerFunc(func(c context.Context, mb service.MessageBatch) error {
+		msgBytes, err := mb[0].AsBytes()
+		require.NoError(t, err)
+		outBatchMut.Lock()
+		outBatches = append(outBatches, string(msgBytes))
+		outBatchMut.Unlock()
+		return nil
+	}))
+
+	streamOut, err = streamOutBuilder.Build()
+	require.NoError(t, err)
+
+	go func() {
+		assert.NoError(t, streamOut.Run(context.Background()))
+	}()
+
+	time.Sleep(time.Second * 5)
+	for i := 0; i < 50; i++ {
+		_, err = db.Exec("INSERT INTO flights (name, created_at) VALUES ($1, $2);", fake.Address().City(), fake.Time().RFC1123(time.Now()))
+		require.NoError(t, err)
+	}
+
+	assert.Eventually(t, func() bool {
+		outBatchMut.Lock()
+		defer outBatchMut.Unlock()
+		return len(outBatches) == 50
+	}, time.Second*20, time.Millisecond*100)
+
+	require.NoError(t, streamOut.StopWithin(time.Second*10))
+	t.Log("All the conditions are met 🎉")
+
+	t.Cleanup(func() {
+		db.Close()
+	})
+}
+
+func TestIntegrationPgCDCForPgOutputPlugin(t *testing.T) {
+	tmpDir := t.TempDir()
+	pool, err := dockertest.NewPool("")
+	require.NoError(t, err)
+
+	var (
+		resource *dockertest.Resource
+		db       *sql.DB
+	)
+
+	resource, db, err = ResourceWithPostgreSQLVersion(t, pool, "16")
+	require.NoError(t, err)
+	require.NoError(t, resource.Expire(120))
+
+	hostAndPort := resource.GetHostPort("5432/tcp")
+	hostAndPortSplited := strings.Split(hostAndPort, ":")
+	password := "l]YLSc|4[i56%{gY"
+
+	require.NoError(t, err)
+
+	fake := faker.New()
+	for i := 0; i < 10; i++ {
+		_, err = db.Exec("INSERT INTO flights (name, created_at) VALUES ($1, $2);", fake.Address().City(), fake.Time().RFC1123(time.Now()))
+		require.NoError(t, err)
+	}
+
+	template := fmt.Sprintf(`
+pg_stream:
+    host: %s
+    slot_name: test_slot_native_decoder
+    user: user_name
+    password: %s
+    port: %s
+    schema: public
+    tls: none
```

Comment:
This test fails because of this. Might be worth rechecking the other tests as well.